gdx/src/com/badlogic/gdx/utils/JsonValue.java

```diff
@@ -83,6 +83,26 @@ public JsonValue (boolean value) {
 		set(value);
 	}
 
+	/** Creates a deep copy of the specific value, except {@link #parent()}, {@link #next()}, and {@link #prev()} are not null. */
+	public JsonValue (JsonValue value) {
+		this(value, null);
+	}
+
+	private JsonValue (JsonValue value, @Null JsonValue parent) {
```

Comment:
Why is the copy constructor with a parent private? Is this just a really narrow use case? Also, if value.child == value, this will infinitely recurse until it StackOverflowErrors. I believe this can also happen for the single-argument copy constructor.

Reply:
It's likely only needed for the copy constructor implementation. We want to copy a node and all its children. For the first node we don't want to set or copy parent, next, or prev but for everything below it we do. I can't imagine needing that elsewhere.

Reply:
Cool, that seems reasonable then.